common/server_room.cpp

```diff
@@ -285,13 +286,36 @@ void Server_Room::say(const QString &userName, const QString &s, bool sendToIsl)
         QString dateTimeString = dateTime.toString();
         chatMessage.set_time(dateTimeString.toStdString());
         chatMessage.set_sender_name(userName.toStdString());
-        chatMessage.set_message(s.simplified().toStdString());
+        chatMessage.set_message(userMessage.simplified().toStdString());
 
         historyLock.lockForWrite();
-        if (chatHistory.size() >= chatHistorySize)
+        if (chatHistory.size() >= chatHistorySize) {
             chatHistory.removeAt(0);
+        }
+
+        chatHistory.push_back(std::move(chatMessage));
+        historyLock.unlock();
+    }
+}
 
-        chatHistory << chatMessage;
+void Server_Room::removeSaidMessages(const QString &userName, int amount, bool sendToIsl)
+{
+    Event_RemoveMessages event;
+    auto stdStringUserName = userName.toStdString();
```

Comment:
any reason for this to be a variable? It's only used once

Reply:
it's used 10 lines down